### PR TITLE
Use sphinx-rtd-theme 1.2.0rc1

### PR DESCRIPTION
**Pull request by benjaoming**

### CODE REVIEW

**Comment by benjaoming** [comment]
@humitos I've updated the pattern so docs are also built if requirements change.

File: .readthedocs.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ build:
       # If there are no changes (git diff exits with 0) we force the command to return with 183.
       # This is a special exit code on Read the Docs that will cancel the build immediately.
       - |
-        if [ "$READTHEDOCS_VERSION_TYPE" = "external" ] && git diff --quiet origin/main -- docs/;
+        if [ "$READTHEDOCS_VERSION_TYPE" = "external" ] && ! ( git diff --quiet origin/main -- docs/ || git diff --quiet origin/main -- requirements/docs.txt || git diff --quiet origin/main -- .readthedocs.yml );
```
Comment (ericholscher) [inline .readthedocs.yml:27]:
```suggestion
        if [ "$READTHEDOCS_VERSION_TYPE" = "external" ] && git diff --quiet origin/main -- docs/ .readthedocs.yml requirements/docs.txt;
```

I think we can just add each of these files in 1 command.

File: docs/user/build-customization.rst [hunk 2]
```diff
@@ -139,12 +139,14 @@ Here is an example that cancels builds from pull requests when there are no chan
        python: "3.11"
      jobs:
        post_checkout:
-         # Cancel building pull requests when there aren't changed in the docs directory.
+         # Cancel building pull requests when there aren't changed in the docs directory or YAML file.
```
Comment (ericholscher) [inline docs/user/build-customization.rst:142]: I updated our docs with a better example here too 👍

File: docs/user/build-customization.rst [hunk 3]
```diff
@@ -139,12 +139,14 @@ Here is an example that cancels builds from pull requests when there are no chan
...
+         # You can add any other files or directories that you'd like here as well,
+         # like your docs requirements file, or other files that will change your docs build.
```
Comment (ericholscher) [inline docs/user/build-customization.rst:144]: I updated our docs with a better example here too 👍

**Review by ericholscher — APPROVED** [review]
This looks good to me, but I haven't reviewed the docs preview heavily 👍

**Comment by ericholscher** [comment]
I did notice the paragraph markers look a little small?

Screenshot 2022-12-16 at 10 51 14 AM

**Comment by benjaoming** [comment]
Oh interesting - they are completely different for me on https://sphinx-rtd-theme.readthedocs.io/en/stable/demo/structure.html -- even the icon!

image

**Review by humitos — APPROVED** [review]

**Comment by benjaoming** [comment]
I can't find anything that has changed, @ericholscher maybe some kind of font change on your system or blocked font URL.

[With sphinx-rtd-theme 1.1.1, docutils 0.17.1, Sphinx 5.3](https://sphinx-rtd-theme.readthedocs.io/en/stable/demo/structure.html):

```
<a class="headerlink" href="#document-section" title="Permalink to this heading"></a>
```

[With sphinx-rtd-theme 1.2.0rc and docutils 0.18 and Sphinx 5.3](https://sphinx-rtd-theme.readthedocs.io/en/latest/demo/structure.html):

```
<a class="headerlink" href="#document-section" title="Permalink to this heading"></a>
```

[With sphinx-rtd-theme 1.2.0rc and docutils 0.18 an Sphinx 6.0b2](https://sphinx-rtd-theme--1385.org.readthedocs.build/en/1385/)

```
<a class="headerlink" href="#document-section" title="Permalink to this heading"></a>
```

**Pull request merged**